src/main/java/com/google/devtools/build/lib/bazel/repository/LocalConfigPlatformFunction.java

```diff
@@ -0,0 +1,160 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.bazel.repository;
+
+import com.google.common.collect.ImmutableList;
+import com.google.devtools.build.lib.analysis.BlazeDirectories;
+import com.google.devtools.build.lib.analysis.RuleDefinition;
+import com.google.devtools.build.lib.packages.Rule;
+import com.google.devtools.build.lib.rules.repository.RepositoryDirectoryValue;
+import com.google.devtools.build.lib.rules.repository.RepositoryDirectoryValue.Builder;
+import com.google.devtools.build.lib.rules.repository.RepositoryFunction;
+import com.google.devtools.build.lib.util.CPU;
+import com.google.devtools.build.lib.util.OS;
+import com.google.devtools.build.lib.vfs.Path;
+import com.google.devtools.build.skyframe.SkyFunction.Environment;
+import com.google.devtools.build.skyframe.SkyFunctionException.Transience;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/** Create a local repository that describes the auto-detected host platform. */
+public class LocalConfigPlatformFunction extends RepositoryFunction {
+
+  @Override
+  public boolean isLocal(Rule rule) {
+    return true;
+  }
+
+  @Override
+  public Class<? extends RuleDefinition> getRuleDefinition() {
+    return LocalConfigPlatformRule.class;
+  }
+
+  @Override
+  public Builder fetch(
+      Rule rule,
+      Path outputDirectory,
+      BlazeDirectories directories,
+      Environment env,
+      Map<String, String> markerData)
+      throws RepositoryFunctionException {
+
+    CPU hostCpu = CPU.getCurrent();
+    OS hostOs = OS.getCurrent();
+
+    List<String> constraints = generateConstraints(hostCpu, hostOs);
```

Comment:
This is a list to open up the possibility of more constraint types going forward?

Reply:
Exactly.